src/recipe.cpp

```diff
@@ -344,6 +362,35 @@ void recipe::finalize()
         container = item::find_type( result_ )->default_container.value_or( "null" );
     }
 
+    std::set<proficiency_id> required;
+    std::set<proficiency_id> used;
+    for( const recipe_proficiency &rpof : proficiencies ) {
+        if( !rpof.id.is_valid() ) {
+            debugmsg( "proficiency %s does not exist in recipe %s", rpof.id.str(), ident_.str() );
+        }
+
+        if( rpof.required && rpof.time_multiplier != 1.0f ) {
+            debugmsg( "proficiencies in recipes cannot be both required and provide a malus in %s",
+                      rpof.id.str(), ident_.str() );
+        }
+        if( required.count( rpof.id ) || used.count( rpof.id ) ) {
+            debugmsg( "proficiency %s both required and provides malus in recipe %s", rpof.id.str(),
+                      ident_.str() );
```

Comment:
Other situations can give rise to this error, like listing a proficiency as required twice.

Reply:
That was partially the intent of this, but that should be a separate check.